lib/spack/spack/cmd/export.py

```diff
@@ -0,0 +1,196 @@
+                           default=None,
+                           help="point to modules generated for MOD",
+                           metavar="MOD")
+    subparser.add_argument("--scope", choices=scopes,
+                           default=spack.config.default_modify_scope(),
+                           help="Configuration scope to modify.")
+    arguments.add_common_arguments(subparser, ['tags', 'constraint'])
+
+
+def export(parser, args):
+    specs = args.specs()
+    # Exit early if no package matches the constraint
+    if not args.specs and args.constraint:
+        msg = "No package matches the query: {0}"
+        msg = msg.format(' '.join(args.constraint))
+        tty.msg(msg)
+        return
+
+    packages = spack.config.get('packages', scope=args.scope)
+
+    # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        specs = [x for x in specs if x.name in packages_with_tags]
+
+    cls = None
+    if args.module:
+        cls = spack.modules.module_types[args.module]
+
+    # Collect packages per package name
+    pkgs = {}
+    for spec in specs:
+        pkgs.setdefault(spec.name, []).append(spec)
+
+    pymods = {}
+
+    # Dump per package, make sure that none are forgotten
+    for pkg, pkg_specs in pkgs.items():
+        paths = syaml_dict()
+        modules = syaml_dict()
+        package = packages.setdefault(pkg, syaml_dict())
+        versions = None
+        if 'version' in package:
+            versions = [str(v) for v in package['version']]
+
+        for spec in pkg_specs:
+            key = spec.format(args.format)
+            sflags = []
+            bflags = []
+            for k, v in spec.variants.items():
+                default = None
+                if k in spec.package.variants:
+                    default = spec.package.variants[k].default
+                if v.value != default:
+                    if v.value in (True, False):
+                        bflags.append(v)
+                    elif v.name != 'patches':
+                        sflags.append(v)
+
+            sflags = ' '.join(str(f) for f in sorted(sflags))
+            bflags = ''.join(str(f) for f in sorted(bflags))
+            key = ' '.join([e for e in (key, sflags, bflags) if len(e) > 0])
+
+            if isinstance(spec.package, PythonPackage):
+                py = spec['python']
+                if args.dependencies:
+                    key += " ^{0}".format(py.format("$_$@"))
+
+                if not spec.package.is_activated(py.package.view()):
+                    # For external packages, setup_environment is not
+                    # called, and thus they are not included in
+                    # PYTHON_PATH.
+                    msg = "python package not activated, skipping: {0}"
+                    msg = msg.format(spec.format("$_$@"))
+                    tty.warn(msg)
```

Comment:
I was trying to see if we can workaround this in #125 but that doesn't capture dependencies. But for now its ok as we will be activating one version.